sdk/scheduler/src/main/java/com/mesosphere/sdk/http/endpoints/HealthResource.java

```diff
@@ -1,79 +1,515 @@
 package com.mesosphere.sdk.http.endpoints;
 
 import com.mesosphere.sdk.http.ResponseUtils;
-import com.mesosphere.sdk.scheduler.SchedulerConfig;
+import com.mesosphere.sdk.scheduler.plan.Plan;
 import com.mesosphere.sdk.scheduler.plan.PlanCoordinator;
-import com.mesosphere.sdk.scheduler.plan.PlanManager;
+import com.mesosphere.sdk.state.FrameworkStore;
+import com.mesosphere.sdk.state.StateStoreException;
+
+import org.apache.mesos.Protos;
+import org.json.JSONArray;
+import org.json.JSONObject;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Response;
 
-import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
 
+// CHECKSTYLE:OFF LineLengthCheck
+
 /**
- * A read-only API for checking whether the service is healthy.
+ * This class returns a single code representing the status of the service
+ * with the following priority.
+ * HTTP Response Code, Priority, Reason
+ * 418, 1, Initializing
+ * 200, 1, Running
+ * 500, 1, Error Creating Service
+ * 204, 2, Deploying:Pending
+ * 202, 2, Deploying:Starting
+ * 206, 3, Degraded
+ * 203, 4, Recovering:Pending
+ * 205, 4, Recovering:Starting
+ * 420, 5, Backing Up
+ * 421, 5, Restoring
+ * 426, 6, Upgrade/Rollback/Downgrade
+ * 503,  , Service Unavailable (Priority Undefined)
  */
 @Path("/v1/health")
 public class HealthResource {
+
   /**
-   * We store the PlanManagers, not the underlying Plans, because PlanManagers can change their plans at any time.
+   * ServiceStatusCode encodes the service status code.
    */
-  private final Collection<PlanManager> planManagers;
+  public enum ServiceStatusCode {
```

Comment:
```suggestion
 enum ServiceStatusCode {
```
is there a downside of making this package-private ?